openapi/openapi.json

```
@@ -32663,6 +32702,26 @@
         }
       }
     },
+    "com.github.openshift.api.operator.v1.Theme": {
+      "description": "Theme defines a theme type for the console UI.",
+      "type": "object",
+      "required": [
+        "type",
+        "file"
+      ],
+      "properties": {
+        "file": {
+          "description": "file is a required field that references a specific file within a ConfigMap in the openshift-config namespace. This field allows the console to locate and use the specified file containing a custom logo. You can create it with a command like: - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file The ConfigMap key should include the file extension for the dark or light or default theme so that the console serves the file with the correct MIME type. Recommended Masthead and About Modal logo specifications: Image size must be less than 1 MB due to constraints on the ConfigMap size Dimensions: Max height of 60px and max width of 200px SVG format preferred Recommended Favicon specifications: Dimensions: Max height of 16px and max width of 16px PNG format preferred",
+          "default": {},
+          "$ref": "#/definitions/com.github.openshift.api.config.v1.ConfigMapFileReference"
+        },
+        "type": {
+          "description": "type is a required field that specifies the type of the logo for the console UI. This field is an enum with valid values \"Dark\", \"Light\" and \"Default\". The type determines whether the logo is for the dark or light or default theme. When set to \"Dark\", the corresponding logo for the dark theme will be used in the console UI. When set to \"Light\", the corresponding logo for the light theme will be used in the console UI. When set to \"Default\", the corresponding logo will be used for any unspecified theme(s) in the console UI. If only the dark or light theme is specified, the default OCP logo will be used for the unspecified theme.",
```

```suggestion
          "description": "type is a required field that specifies the type of logo for the console UI. This field is an enum with valid values \"Dark\", \"Light\" and \"Default\". The type determines whether the logo is for the dark or light or default theme. When set to \"Default\", the corresponding logo will be used for any unspecified theme(s) in the console UI. If only the dark or light theme is specified, the default OCP logo will be used for the unspecified theme.",
```

same comment as above for enum.

Also wondering if these 2 sentences are necessary. It is already implied with `The type determines whether the logo is for the dark or light or default theme.` imo. The default statement should stay though because it needs explanation.
`When set to \"Dark\", the corresponding logo for the dark theme will be used in the console UI. When set to \"Light\", the corresponding logo for the light theme will be used in the console UI. `

---

Good point, but I wanted to be more explicit as suggested by one of the reviewers, and stated in the [API docs](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc)